R/digest.R

```diff
@@ -54,10 +53,10 @@ digest <- function(object, algo=c("md5", "sha1", "crc32", "sha256", "sha512",
         file <- TRUE                  	# nocov
     }
 
-    is_streaming_algo <- algo == "spookyhash"
+    is_streaming_algo <- algoint == 9L
```

Comment:
That's trickier.  The is time in code execution, and there is clarity in code.  This no longer wins.

(We could be tricky and make it a factor.  Explicit, easy to read, still pass an integer down.)

Reply:
can do it as algo[1] comparison instead?

Reply:
(i *think* the reason the `algo <- algo[1]` reassignment was slowing things down is triggering when R treats function arguments as read-only vs not, and inadvertantly triggering a copy operation?)